test/enums.jl

```diff
@@ -0,0 +1,130 @@
+module TestEnums
+
+using Base.Test
+
+@test_throws MethodError convert(Base.Enums.Enum, 1.0)
+
+Base.Enums.@enum Fruit apple orange kiwi
```

Comment:
It actually is exported from Base in this PR, I just tend to write tests as fully qualified as possible :)